js/ui/editors/ConstructorControlPanel.js

```diff
@@ -319,20 +324,71 @@ export class ConstructorControlPanel extends ConstructorEditorView {
             projectProperties.config_name_set
         );
 
+        UIPreferences.addDropdown(
+            this.projectSettings,
+            'Runner Type',
+            projectProperties.runner_project_get() ?? 'Use Default',
+            [...preferences.valid_runner_types, 'Use Default'],
+            (value) => {
+
+                if (value === 'Use Default') {
+                    projectProperties.runner_set(undefined);
+                    return;
+                }
+
+                // @ts-ignore
+                projectProperties.runner_set(value);
+            }
+        );
+
         UIPreferences.addDropdown(
             this.projectSettings,
             'Runtime Channel Type',
-            projectProperties.runtime_channel_type_get(),
+            projectProperties.runtime_project_channel_type_get() ?? 'Use Default',
             [...preferences.valid_runtime_types, 'Use Default'],
             (value) => {
 
                 if (value === 'Use Default') {
```

Comment:
When the callbacks are getting this long I'd prefer they're proper callbacks on the class itself